### ISSUE TITLE
.NET Core 2.0 and netstandard 2.0 support

### ISSUE THREAD

**Issue by denizalpaslan** [Antaris/RazorEngine#515]
Hi,

**TL;DR**
Does this library support netstandard2.0 ? I'm trying to use RazorEngine 3.10.0

**Details**
I have a .NET core 2.0 project (Project A) which has a reference to another project with netstandard 2.0. (Project B)

I'm trying to use RazorEngine in Project B : 

- I added package with "dotnet add package RazorEngine". It succeeded with 3.10.0 with a warning that it may not be fully compatible with my project.

- I wrote the code and it builds successfully. When I try to run it I got "Could not load file or assembly 'System.Configuration.ConfigurationManager ...." error and I added this package reference explicitly (saw another issue about that before)

- The second time I ran the code, I got "Could not load file or assembly 'System.Web.Razor ..." which doesn't seem to be part of netstandard2.0

Any idea?

**Comment by stusklinar** [comment]
Try adding the microsoft.aspnet.razor package to get System.Web.Razor

**Comment by stusklinar** [comment]
Also see:

#498

**Comment by denizalpaslan** [comment]
Nope. It now gives 

`Exception="System.TypeLoadException" ExceptionDetail="System.TypeLoadException: Could not load type 'Microsoft.AspNetCore.Razor.Parser.ParserBase' from assembly 'Microsoft.AspNetCore.Razor, Version=2.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'`

which is caused by a breaking change introduced with .NET core 2.0

**Comment by stusklinar** [comment]
That's because razor 2.0 isn't supported, I'm also waiting on a fix.

**Comment by daviatorstorm** [comment]
+1

**Comment by denizalpaslan** [comment]
@matthid  & @conniey any timeline for this update? I'm mentioning you because I see you guys as the top two contributors.

Thanks

**Comment by stusklinar** [comment]
Ok - Update.

There *is* a NetCore release for this - https://github.com/hermanho/RazorEngine/tree/updateToNetCore

This is forked from this branch, which was then forked from @conniey.

Available as a nuget package here: 

https://www.nuget.org/packages/RazorEngine.Core.NetCore/

Now - whether you use this in prod - is *totally up to you*, I'd advise a lot of testing!

If you use a base template that doesn't have a empty ctor, this code won't work - it just won't. I've raised #517 for this.

If you use a normal base template, crack on.

**Comment by mscrivo** [comment]
I tried the package that @stusklinar suggested and while it compiles, it doesn't seem to work at all with my existing templates. 

This dependency on Microsoft.AspNetCore.Razor 1.x is becoming really problematic because it has a dependency on System.Net.Http nuget package which completely breaks things if you target .NET 4.7.2 and also have projects that target .NET Standard and have code that uses System.Net.Http.   The above package resolves those build issues, but as mentioned, does not work.

**Comment by stusklinar** [comment]
What are your existing templates? Can you show me an example? I May have a fix if you have an extended base template.

**Comment by mscrivo** [comment]
@stusklinar 
Here's the error I'm getting: 

```
RazorEngine.Templating.TemplateCompilationException: 'Errors while compiling a Template.
 - error: (25, 6) The name 'inherits' does not exist in the current context
```

The template itself looks like this:

```
@using MyNameSpace
@inherits RazorEngine.Templating.TemplateBase<ModelClass>
@{
    const int maxColsPerRow = 5;
}
<tr>
A bunch of standard razor code here
</tr>
```
The class `ModelClass` is a fairly standard class, but it inherits from a base class, which implements an interface.

**Comment by stusklinar** [comment]
Ah ok, different error to mine.

So long as your base classes don't need ctor params, you won't run into the same issue me.

As for this, can you pass your model inherit this instead?

**Comment by mscrivo** [comment]
> As for this, can you pass your model inherit this instead?

Not sure what you mean by that?

**Comment by stusklinar** [comment]
Well, as you're having a base template, in the razor config you supply to the engine, you can specify a base template class, could you supply that instead??

`/// <summary>
		/// The type of the view base to compile templates with
		/// </summary>
		public override Type ViewBaseType
		{
			get { return typeof(ViewBase<>); }
		}`

**Comment by mscrivo** [comment]
Sorry, still not sure what you're referring to? ... I don't see that ViewBaseType property anywhere.  Are you referring to `BaseTemplateType`?  The problem is, I don't have a base type for my actual templates, only the models have a common interface.

The way I'm calling the Engine is like this:

`var emailHtml = Engine.Razor.RunCompile(templateName, null, emailModel);`

The engine is setup like this:
```
var config = new TemplateServiceConfiguration
{
#if !DEBUG
    // This allows Razor Engine to clean up after itself. 
    DisableTempFileLocking = true,
#endif
    TemplateManager = new ResolvePathTemplateManager(new[] {Path.Combine(AssemblyDirectory, @"Reporting\RazorEmailTemplates")}),

    // Enable Roslyn compiler for templates to get C# 6+ features
    CompilerServiceFactory = new RoslynCompilerServiceFactory(),
#if DEBUG
    // enable debugging in templates
    Debug = true
#endif
};


var service = RazorEngineService.Create(config);

Engine.Razor = service;
```

**Comment by stusklinar** [comment]
I think they're the same thing aren't they? The base template has a model on it. And if you inherit from it, you can set your model in that?

**Comment by kralikba** [comment]
Hey,
I'm having the same problem. 
Both `@model` and `@inherits` directives end up parsed as simple expressions instead of directives - i.e.
the following is generated:
``` Write(model); ```

**Comment by mariusene** [comment]
I have the same issue as @kralikba :  @inherits directives are parsed as simple expressions.
Any fix for this?

**Comment by janniksam** [comment]
Having the same issue.

I have to comment out the inherits-line when releasing my razor-views and comment it in again to design the views.

A workaround for me would be to replace the inherits line before sending it into the RazorEngine, but thats a huge hack imho.

**Comment by janniksam** [comment]
Temporary workaround (apply before compiling):
```cs
var regex = new Regex("^(\\s*@inherits.*)$", RegexOptions.Multiline);
parsableTemplate = regex.Replace(parsableTemplate, string.Empty);
```

**Comment by one-matrix** [comment]
hope support netcore

**Comment by Zetanova** [comment]
any news?
